### PR TITLE
bpo-42487: don't call __getitem__ of underlying maps in ChainMap.__iter__

### PR DESCRIPTION
**Pull request by ap--**
See: https://bugs.python.org/issue42487

https://bugs.python.org/issue42487

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@ap--

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

File: Lib/collections/__init__.py [hunk 1]
```diff
@@ -1001,7 +1001,7 @@ def __len__(self):
     def __iter__(self):
         d = {}
         for mapping in reversed(self.maps):
-            d.update(mapping)                   # reuses stored hash values if possible
+            d.update(dict.fromkeys(mapping))    # reuses stored hash values if possible
```
Comment (rhettinger) [inline Lib/collections/__init__.py:1004]: *dict.fromkeys()* keeps the keys but loses the values.

**Review by rhettinger — CHANGES_REQUESTED** [review]

**Review by rhettinger — CHANGES_REQUESTED** [review]
Please add tests and a blurb entry for Misc/NEWS.  Also, please get the CLA signed.

**Comment by ap--** [comment]
Hi @rhettinger,
I added a test and the news blurb and signed the CLA (though it states on bpo it make take up to few days to register...)

Thanks for the quick review and feedback :)

Cheers,
Andreas

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

**Review by rhettinger — APPROVED** [review]

**Pull request merged**